src/components/AudioPlayer/getHighlightedLocation.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  HighlightedLocationState,
+  initialState as defaultHighlightedLocation,
+} from 'src/redux/slices/QuranReader/highlightedLocation';
+import { getVerseAndChapterNumbersFromKey } from 'src/utils/verse';
+import VerseTiming from 'types/VerseTiming';
+
+/**
+ * given the currentTime of the audio player and the verseTimings
+ * determine which `chapter`, `verse`, and `word` should currently be highlighted
+ *
+ * @param {number} currentTime
+ * @param {VerseTiming[]} verseTimings
+ * @returns {HighlightedLocationState} highlightedLocation
+ */
+const getHighlightedLocation = (
+  currentTime: number,
+  verseTimings: VerseTiming[],
+): HighlightedLocationState => {
+  const selectedVerse = verseTimings.find(
+    (verse) => currentTime >= verse.timestampFrom && currentTime <= verse.timestampTo,
+  );
+
+  if (!selectedVerse) return defaultHighlightedLocation;
+  const selectedWord = selectedVerse.segments.find((segment) => {
+    const [, timestampFrom, timestampTo] = segment; // the structure of the segment is: [wordLocation, timestampFrom, timestampTo]
+    return currentTime >= timestampFrom && currentTime <= timestampTo;
+  });
+
+  if (!selectedWord) return defaultHighlightedLocation;
```

Comment:
this is what needs to change, handle the case where the verse is highlighted but there's no word to highlight.